### PR TITLE
Ticket 4

### PR DESCRIPTION
**Pull request by benguimo**
Ticket 4.  Show all articles.
Ticket 5 was done while correcting and adding loading/error handling, to test single article as well as all articles.

### CODE REVIEW

**Review by mezzDavies — CHANGES_REQUESTED** [review]
There is a lot going on here that is above and beyond the ticket instructions. Really we should stick to what the ticket/task asks of us. That said I can see where you are going and I appreciate your enthusiasm! Just try to do only what is asked.


Something that you need to take care of please though is the loading state. We should consider what the user sees if for example the connection is very slow and it is taking time for the articles to load. In other words we need some loading state to conditionally provide feedback whilst we wait for the asynchronous task of getting the articles, so the user isn't just staring at a blank screen wondering what is going on!

Please can you implement this and resubmit the PR. 👍 

Also your CSS may need a tweak as this is how the home page looks for me:

image

File: src/components/Articles.jsx [hunk 1]
```diff
@@ -12,11 +12,20 @@ function Articles() {
   const [articles, setArticles] = useState([]);
   const [search, setSearch] = useState("");
   const [topics, setTopics] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false)
 
   useEffect(() => {
+    setLoading(true);
+    setError(false);
```
Comment (Beth-McCormack) [inline src/components/Articles.jsx]: The error is already set to false on line 16 so this line is not needed🙂

File: src/components/Articles.jsx [hunk 2]
```diff
@@ -12,11 +12,20 @@ function Articles() {
...
     getArticles().then((res) => {
       const newList = res.filter((article) => article.title.toLowerCase().includes(search.toLowerCase()));
```
Comment (Beth-McCormack) [inline src/components/Articles.jsx:19]: This is not currently needed for this ticket, and the user experience for the search bar is not great currently. If you want to implement this now, you need to make sure its working well. I would personally delete this and you can add it when you get to the advanced tickets.

File: src/components/Articles.jsx [hunk 3]
```diff
@@ -1,11 +1,90 @@
...
+  const [articles, setArticles] = useState([]);
+  const [search, setSearch] = useState("");
+  const [topics, setTopics] = useState([]);
+
+  useEffect(() => {
+    getArticles().then((res) => {
+      const newList = res.filter((article) => article.title.toLowerCase().includes(search.toLowerCase()));
+      setArticles(newList);
+    });
+    getTopics().then((res) => {
```
Comment (Beth-McCormack) [inline src/components/Articles.jsx]: Topics are not dealt with until ticket 9. Please delete this and only add functionality that is needed for the specific ticket you are doing. Not only will it take you longer to do but will also me more difficult for us to review🙂

File: src/components/Articles.jsx [hunk 4]
```diff
@@ -1,11 +1,90 @@
...
+      setTopics(res);
+    });
+  }, [search]);
+
+
+  useEffect(() => {
+    getTopics().then((res) => {
```
Comment (Beth-McCormack) [inline src/components/Articles.jsx]: See above

File: src/components/Articles.jsx [hunk 5]
```diff
@@ -1,11 +1,90 @@
...
+      setTopics(res);
+    });
+  }, []);
+
+  function chooseTopic(e) {
+    getArticles().then((res) => {
+      const sortedBy = res.filter((article) => {
+        return article.topic === e.target.innerText;
+      });
+      setArticles(sortedBy);
```
Comment (Beth-McCormack) [inline src/components/Articles.jsx]: See above

**Review by Beth-McCormack — APPROVED** [review]
Nice work so far. Your CSS is fixed and looks good! 
I think you've done too much for this ticket and have implemented features that are not needed for quite a few more tickets. Please only commit code which is relevant to the ticket you are doing😁 Please have a look through my comments and act on them before you merge

**Pull request merged**